public/types.ts

```diff
@@ -31,6 +34,7 @@ export interface SecurityPluginSetupDependencies {
   savedObjectsManagement: SavedObjectsManagementPluginSetup;
   managementOverview?: ManagementOverViewPluginSetup;
   dataSourceManagement?: DataSourceManagementPluginSetup;
+  dataSource?: DataSourcePluginSetup;
```

Comment:
This is for checking if mds is enabled or not right?

Reply:
Yea, but that was to address some type check that was unrelated to the change. I reverted it and will follow up. I limited the PR just to the file which needs touching and adding a test file.